### PR TITLE
DTTool: ouput columns order by column id

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/File/DMFile.h [hunk 1]
```diff
@@ -299,7 +299,7 @@ class DMFile : private boost::noncopyable
     DMConfigurationOpt & getConfiguration() { return configuration; }
 
     /**
-     * Return all column defines. This is useful if you want to read all data from a dmfile.
+     * Return all column defines sort by column id. This is useful if you want to read all data from a dmfile.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/File/DMFile.h:302]: maybe make `sorted` as an input param and default to false? Sometimes we don't need it to be sorted.

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/File/DMFile.h:302]: In which case do need to be sorted?

File: dbms/src/Storages/DeltaMerge/File/DMFile.h [hunk 2]
```diff
@@ -303,14 +303,18 @@ class DMFile : private boost::noncopyable
...
+    ColumnDefines getColumnDefines(bool sort_by_id = true)
     {
         ColumnDefines results{};
         results.reserve(this->column_stats.size());
         for (const auto & cs : this->column_stats)
         {
             results.emplace_back(cs.first, "", cs.second.type);
         }
+        if (sort_by_id)
+            std::sort(results.begin(), results.end(), [](const auto & lhs, const auto & rhs) {
```
Comment (CalvinNeo) [inline dbms/src/Storages/DeltaMerge/File/DMFile.h:315]: Is it better to use cbegin and cend?

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/File/DMFile.h:315]: no, sort will change the vector, can not use const iterator.

File: dbms/src/Storages/DeltaMerge/File/DMFile.h (hunk 1 above)
Comment (JaySon-Huang) [reply dbms/src/Storages/DeltaMerge/File/DMFile.h]: We don't care whether it is sorted or not in its previous usage. But the sorted is needed to the bloomfilter POC

**Review by JaySon-Huang — APPROVED** [review]

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**